test/common/testlib.py

```diff
@@ -1123,10 +1123,34 @@ def check_journal_messages(self, machine=None):
         machine = machine or self.machine
         # on main machine, only consider journal entries since test case start
         cursor = (machine == self.machine) and self.journal_start or None
-        syslog_ids = ["cockpit-ws", "cockpit-bridge"]
+
+        # Journald does not always set trusted fields like
+        # _SYSTEMD_UNIT or _EXE correctly for the last few messages of
+        # a dying process, so we filter by the untrusted but reliable
+        # SYSLOG_IDENTIFIER instead.
+
+        def syslog_id(v):
+            return "SYSLOG_IDENTIFIER=" + v
+
+        def glib_domain(v):
+            return "GLIB_DOMAIN=" + v
+
+        matches = [
+            syslog_id("cockpit-ws"),
+            syslog_id("cockpit-bridge"),
+            syslog_id("cockpit/ssh"),
+            glib_domain("cockpit-ws"),
+            glib_domain("cockpit-bridge"),
+            glib_domain("cockpit-ssh"),
+            glib_domain("cockpit-pcp")
+        ]
+
         if not self.allow_core_dumps:
-            syslog_ids += ["systemd-coredump"]
-        messages = machine.journal_messages(syslog_ids, 6, cursor=cursor)
+            matches += [syslog_id("systemd-coredump")]
+
+        messages = machine.journal_messages(matches, 6, cursor=cursor)
+        print("M", messages)
```

Comment:
debugging, remove before landing

Reply:
I leave it in for a bit for sanity checking, it is so easy to mess up the match expressions and end up with no journal messages at all...

Reply:
Removed